src/main/kotlin/graphics/scenery/backends/vulkan/VulkanDevice.kt

```diff
@@ -362,14 +362,28 @@ open class VulkanDevice(
         vkDestroySemaphore(this.vulkanDevice, semaphore, null)
     }
 
-    data class DescriptorPool(val handle: Long, var free: Int = maxTextures + maxUBOs + maxInputAttachments + maxUBOs) {
+    data class DescriptorPool(
+        val handle: Long,
+        var free: ConcurrentHashMap<Int, Int> = ConcurrentHashMap(
```

Comment:
Does it really need to be `Concurrent`?

Reply:
Yes, as  it might be accessed by different threads, at least for reading. This is to ensure consistency.